astropy/coordinates/tests/test_earth.py

```diff
@@ -7,16 +7,43 @@
 
 import pytest
 import numpy as np
+from numpy.testing import assert_allclose
 
-from astropy.coordinates.earth import EarthLocation, ELLIPSOIDS
+from astropy.coordinates.representation import CartesianRepresentation, REPRESENTATION_CLASSES
+from astropy.coordinates.earth import EarthLocation, ELLIPSOIDS, WGS84GeodeticRepresentation
 from astropy.coordinates.angles import Longitude, Latitude
 from astropy.units import allclose as quantity_allclose
+from astropy.utils import isiterable
 from astropy import units as u
 from astropy.time import Time
 from astropy import constants
 from astropy.coordinates.name_resolve import NameResolveError
 
 
+def components_equal(rep1, rep2):
+    result = True
+    if type(rep1) is not type(rep2):
+        return False
+    for component in rep1.components:
+        result &= getattr(rep1, component) == getattr(rep2, component)
+    return result
+
+
+def representation_equal(rep1, rep2):
+    result = True
+    if type(rep1) is not type(rep2):
+        return False
+    if getattr(rep1, '_differentials', False):
+        if rep1._differentials.keys() != rep2._differentials.keys():
+            return False
+        for key, diff1 in rep1._differentials.items():
+            result &= components_equal(diff1, rep2._differentials[key])
+    elif getattr(rep2, '_differentials', False):
+        return False
+
+    return result & components_equal(rep1, rep2)
```

Comment:
These are copy-pasted from `test_representations.py`. Should we move them to `astropy.utils` or `astropy.tests.helper`, or is it not too bad to have them duplicated?

Reply:
I'd just import then from `test_representation` for now (also done in `test_representation_methods`...)

Reply:
Done!